### PR TITLE
Run2-hcx62 Validation toolket for HGCAL

### PR DESCRIPTION
**Pull request by bsunanda**
This PR can be tested only when #12845 is integrated. It uses many of the new functionalities there

### CODE REVIEW

**Comment by lgray** [comment]
@cmsbuild please test

Dependencies are now in CMSSW.

File: Validation/HGCalValidation/plugins/HGCalDigiValidation.cc [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    iSetup.get<IdealGeometryRecord>().get(nameDetector_, geom);
+    if (!geom.isValid()) edm::LogWarning("HGCalValidation") << "Cannot get valid HGCalGeometry Object for " << nameDetector_;
+    geom0 = geom.product();
+  }
+
+  unsigned int ntot(0), nused(0);
+  if (nameDetector_ == "HGCalEESensitive") {
+    //HGCalEE
+    edm::Handle<HGCEEDigiCollection> theHGCEEDigiContainers;
+    iEvent.getByLabel(digiSource_, theHGCEEDigiContainers);
```
Comment (deguio) [inline Validation/HGCalValidation/plugins/HGCalDigiValidation.cc]: this needs to be migrated to consumes

File: Validation/HGCalValidation/plugins/HGCalDigiValidation.cc [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+      }
+      fillDigiInfo();
+    } else {
+      edm::LogWarning("HGCalValidation") << "HGCEEDigiCollection handle does not exist !!!";
+    }
+  } else if ((nameDetector_ == "HGCalHESiliconSensitive") || 
+	     (nameDetector_ == "HGCalHEScintillatorSensitive")) {
+    //HGCalHE
+    edm::Handle<HGCHEDigiCollection> theHGCHEDigiContainers;
+    iEvent.getByLabel(digiSource_, theHGCHEDigiContainers);
```
Comment (deguio) [inline Validation/HGCalValidation/plugins/HGCalDigiValidation.cc]: this needs to be migrated to consumes

File: Validation/HGCalValidation/plugins/HGCalDigiValidation.cc [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+	digiValidation(detId, geom0, layer, adc, charge);
+      }
+      fillDigiInfo();
+    } else {
+      edm::LogWarning("HGCalValidation") << "HGCHEDigiCollection handle does not exist !!!";
+    }
+  } else if (nameDetector_ == "HCal") {
+    //HE
+    edm::Handle<edm::SortedCollection<HcalUpgradeDataFrame> >  theHEDigiContainers;
+    iEvent.getByLabel(digiSource_, theHEDigiContainers);
```
Comment (deguio) [inline Validation/HGCalValidation/plugins/HGCalDigiValidation.cc]: this needs to be migrated to consumes

File: Validation/HGCalValidation/plugins/HGCalRecHitValidation.cc [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+  bool ok(true);
+  unsigned int ntot(0), nused(0);
+  if (nameDetector_ == "HCal") {
+    edm::ESHandle<CaloGeometry> geom;
+    iSetup.get<CaloGeometryRecord>().get(geom);
+    if (!geom.isValid()) edm::LogWarning("HGCalValidation") << "Cannot get valid HGCalGeometry Object for " << nameDetector_;
+    const CaloGeometry* geom0 = geom.product();
+
+    edm::Handle<HBHERecHitCollection> hbhecoll;
+    iEvent.getByLabel(recHitSource_, hbhecoll);
```
Comment (deguio) [inline Validation/HGCalValidation/plugins/HGCalRecHitValidation.cc]: this needs to be migrated to consumes

File: Validation/HGCalValidation/plugins/HGCalRecHitValidation.cc [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+      edm::LogWarning("HGCalValidation") << "HBHERecHitCollection Handle does not exist !!!";
+    }
+  } else {
+    edm::ESHandle<HGCalGeometry> geom;
+    iSetup.get<IdealGeometryRecord>().get(nameDetector_, geom);
+    if (!geom.isValid()) edm::LogWarning("HGCalValidation") << "Cannot get valid HGCalGeometry Object for " << nameDetector_;
+    const HGCalGeometry* geom0 = geom.product();
+
+    edm::Handle<HGCRecHitCollection> theRecHitContainers;
+    iEvent.getByLabel(recHitSource_, theRecHitContainers);
```
Comment (deguio) [inline Validation/HGCalValidation/plugins/HGCalRecHitValidation.cc]: this needs to be migrated to consumes

**Comment by lgray** [comment]
@deguio let me know when you're done reviewing and I'll address your comments, and open a new PR since Sunanda is away.

**Comment by deguio** [comment]
ciao @lgray 
I think this one can go in once the consumes migration is addressed. this PR has been reviewed back in Oct/Nov (see #11872).
for reference, here the details presented and discussed at the DQM meeting:
https://indico.cern.ch/event/453276/contribution/3/attachments/1169232/1687152/Phase2-Talk22.pdf

ciao and thanks,
F.

**Comment by lgray** [comment]
OK -- it is added to the stack. Need to find a stopping point in trying to debug hex geometry.

**Comment by lgray** [comment]
-1 I will open a new PR for this with the requested changed momentarily.

**Comment by lgray** [comment]
@davidlange6 please close this PR. Thanks!

**Comment by deguio** [comment]
-1
superseded by #13046

**Comment by deguio** [comment]
+1

**Comment by lgray** [comment]
@deguio wrong place for the +1 :)

**Comment by deguio** [comment]
-1

**Pull request merged**